File: arcade/0003-checkPalindrome/checkPalindrome.cpp
bool checkPalindrome(const std::string& s) {
    int length = s.length();
    int halfLength = length / 2;
    
    for (int i = 0; i < halfLength; ++i) {
        if (s[i] != s[length - i - 1])
            return false;
    }
    
    return true;
}

File: arcade/0003-checkPalindrome/checkPalindrome.kt
fun checkPalindrome(inputString: String): Boolean {
  val length = inputString.length 
  val halfLength = length / 2
  for (i in 0 until halfLength) {
      if (inputString[i] != inputString[length - i - 1]) {
          return false
      }
  }
  return true
}